lib/simplehttp.ex

```diff
@@ -30,12 +30,22 @@ defmodule SimpleHttp do
   @spec request(atom(), String.t(), keyword()) :: {:error, any()} | {:ok, SimpleHttp.Response.t()}
   def request(method, url, args \\ []) do
     request = create_request(method, url, args)
+    if request.global_options != [] do
```

Comment:
from what I understood global opts are used just for initialization of httpc, in my opinion could should not be part of request. I would go with something like
```
def request(method, url, args \\ []) do
    initialize_httpc_opts(args)
    request = create_request(method, url, args)
    execute_requesst()
end
```
what do you think ?

Reply:
I agree and reworked this and also added support for optionally running requests in custom profiles.